src/v2i-hub/PedestrianPlugin/src/FLIRWebSockAsyncClnSession.cpp

```diff
@@ -0,0 +1,370 @@
+    FLIRWebSockAsyncClnSession::on_read(
+        beast::error_code ec,
+        std::size_t bytes_transferred)
+    {
+        PLOG(logDEBUG) << "In FLIRWebSockAsyncClnSession::on_read " << std::endl;
+        boost::ignore_unused(bytes_transferred);
+
+        if(ec)
+            return fail(ec, "read");
+
+        // parse the buffer
+        std::stringstream ss; 
+	    ptree pr; 
+	    std:string text(beast::buffers_to_string(buffer_.data()));
+        ss<<text; 
+
+        PLOG(logDEBUG) << "Received:  " << text.c_str() << std::endl;
+
+        try
+        {
+	        read_json(ss, pr);
+        }
+        catch(const ptree_error &e)
+        {
+            PLOG(logERROR) << "Error converting json to p tree:  " << e.what() << std::endl;
+        }
+
+        //Example FLIR subscription response json:
+        //Received:  {"messageType": "Subscription", "subscription": {"returnValue": "OK", "type": "Data"}}
+
+        std::string messageType = pr.get_child("messageType").get_value<string>();
+        std::string time = "";
+        std::string type = "";
+        float angle = 0;
+        float alpha = 0;
+        float lat = 0;
+        float lon = 0;
+        float speed = 0;        
+        std::string timeString = "";
+        int id = 0;
+        std::string idResult;
+
+        if (messageType.compare("Subscription") == 0)
+        {
+            std::string subscrStatus = pr.get_child("subscription").get_child("returnValue").get_value<string>();
+            PLOG(logDEBUG) << "Ped presence data subscription status: " << subscrStatus << std::endl;
+        
+        }
+        //Example received pedestrian tracking data       
+        //{"dataNumber": "473085", "messageType": "Data", "time": "2022-04-20T15:25:51.001-04:00", 
+        //"track": [{"angle": "263.00000000", "class": "Pedestrian", "iD": "15968646", "latitude": "38.95499217", 
+        //"longitude": "-77.14920953", "speed": "1.41873741", "x": "0.09458912", "y": "14.80903757"}], "type": "PedestrianPresenceTracking"}
+        else if (messageType.compare("Data") == 0)
+        {
+            time = pr.get_child("time").get_value<string>(); 
+            type =  pr.get_child("type").get_value<string>();
+
+            PLOG(logINFO) << "Received " << type << " data at time: " << time << std::endl;
+
+            if (type.compare("PedestrianPresenceTracking") == 0)
+            {
+                try
+                {
+                    for (auto it: pr.get_child("track")) 
+                    {                        
+
+                        if (!it.second.get_child("angle").data().empty())
+                        {
+                            angle = std::stof(it.second.get_child("angle").data());
+                            //convert camera reference frame angle
+                            alpha = cameraRotation_ - angle - 270;
+
+                            while (alpha < 0)
+                            {
+                                alpha = 360 + alpha;
```

Comment:
maybe use (alpha % 360) + 360 instead of a while loop?

Reply:
Done